lib/plugins/logviewer/admin.php

```diff
@@ -150,5 +123,86 @@ protected function getFacilities()
         return $facilities;
     }
 
+    /**
+     * Read the lines of the logfile and return them as an array
+     *
+     * @param string $logfilePath
+     * @return array
+     */
+    protected function getLogLines($logfilePath)
+    {
+        global $lang;
+
+        $lines = [];
+        $size = filesize($logfilePath);
+        
+        if ($size <= self::MAX_READ_SIZE) {
+            $lines = file($logfilePath);
```

Comment:
I'm not a fan of having two different method for reading the file. Why not always use fopen and only seek to different positions, depending on the filesize?

Reply:
I guess the reason I separated those was I felt they do different things based on their names and I didn't want to have a `display` method that does both `reading` and `printing` which are different tasks.
I can change if you want but I feel it's a little more clear this way for anyone in future who will possibly wants to change the code.
But your point is also valid because now the code shorted than what I anticipated, so just let me know.

Reply:
No. That's not what I'm talking about. I am talking about using `file()` for reading small log files and `fopen()` for larger ones. Both should be handled by the same fopen/fread code.